src/checkout/views/Billing/View.tsx

```diff
@@ -73,61 +73,97 @@ const computeMutationVariables = (
   }
 });
 
-const View: React.FC<RouteComponentProps<{ token?: string }>> = ({
-  history,
-  match: {
-    path,
-    params: { token }
+class View extends React.Component<
+  RouteComponentProps<{ token?: string }>,
+  { sameAsShipping: boolean; validateStep: boolean }
+> {
+  state = { sameAsShipping: false, validateStep: true };
+
+  componentDidMount() {
+    this.setState({ validateStep: false });
+  }
+
+  render() {
+    const {
+      history,
+      match: {
+        path,
+        params: { token }
+      }
+    } = this.props;
+    const { validateStep, sameAsShipping } = this.state;
+
+    return (
+      <CheckoutContext.Consumer>
+        {({ checkout, update, step }) =>
+          validateStep ? (
+            <StepCheck
+              step={step}
+              checkout={checkout}
+              path={path}
+              token={token}
+            />
+          ) : (
+            <CartSummary checkout={checkout}>
+              <Steps
+                step={CheckoutStep.BillingAddress}
+                token={token}
+                checkout={checkout}
+              >
+                <TypedUpdateCheckoutBillingAddressMutation
+                  onCompleted={data =>
+                    proceedToPayment(data, update, history, token)
+                  }
+                >
+                  {(saveBillingAddress, { data, loading }) => (
+                    <ShopContext.Consumer>
+                      {shop => (
+                        <>
+                          <div className="address-form__copy-address">
+                            <label className="checkbox">
+                              <input
+                                name="asBilling"
+                                type="checkbox"
+                                checked={sameAsShipping}
+                                onChange={({ target: { checked } }) =>
+                                  this.setState({ sameAsShipping: checked })
+                                }
+                              />
+                              <span>Same as Shipping Address</span>
+                            </label>
+                          </div>
+                          <ShippingAddressForm
+                            key={`${sameAsShipping}`}
+                            buttonText="Continue to Payment"
```

Comment:
Maybe it would be better to say "Proceed" instead of "Continue"? I'm not sure. 
```suggestion
                            buttonText="Proceed to Payment"
```